x-pack/plugins/ml/public/application/datavisualizer/index_based/components/expanded_row/geo_point_content.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+
+import { EuiFlexItem } from '@elastic/eui';
+import { ExamplesList } from '../../../index_based/components/field_data_row/examples_list';
+import { FieldVisConfig } from '../../../stats_table/types';
+import { IndexPattern } from '../../../../../../../../../src/plugins/data/common/index_patterns/index_patterns';
+import { MlEmbeddedMapComponent } from '../../../../components/ml_embedded_map';
+import { ML_JOB_FIELD_TYPES } from '../../../../../../common/constants/field_types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { CreateLayerDescriptorParams } from '../../../../../../../maps/public/classes/sources/es_search_source';
+import { ES_GEO_FIELD_TYPE } from '../../../../../../../maps/common/constants';
+import { LayerDescriptor } from '../../../../../../../maps/common/descriptor_types';
+import { useMlKibana } from '../../../../contexts/kibana';
+import { DocumentStatsTable } from '../../../stats_table/components/field_data_expanded_row/document_stats';
+import { ExpandedRowContent } from '../../../stats_table/components/field_data_expanded_row/expanded_row_content';
+
+export const GeoPointContent: FC<{
+  config: FieldVisConfig;
+  indexPattern: IndexPattern | undefined;
+  combinedQuery: { searchString: string; searchQueryLanguage: string };
+}> = ({ config, indexPattern, combinedQuery }) => {
+  const { stats } = config;
+  const [layerList, setLayerList] = useState<LayerDescriptor[]>([]);
+  const {
+    services: { maps: mapsPlugin },
+  } = useMlKibana();
+
+  // Update the layer list  with updated geo points upon refresh
+  useEffect(() => {
+    async function updateIndexPatternSearchLayer() {
+      if (
+        indexPattern?.id !== undefined &&
+        config !== undefined &&
+        config.fieldName !== undefined &&
+        config.type === ML_JOB_FIELD_TYPES.GEO_POINT
+      ) {
+        const params: CreateLayerDescriptorParams = {
+          indexPatternId: indexPattern.id,
+          geoFieldName: config.fieldName,
+          geoFieldType: config.type as ES_GEO_FIELD_TYPE.GEO_POINT,
+          query: {
+            query: combinedQuery.searchString,
+            language: combinedQuery.searchQueryLanguage,
+          },
+        };
+        const searchLayerDescriptor = mapsPlugin
+          ? await mapsPlugin.createLayerDescriptors.createESSearchSourceLayerDescriptor(params)
+          : null;
+        if (searchLayerDescriptor) {
+          setLayerList([...layerList, searchLayerDescriptor]);
+        }
+      }
+    }
+    updateIndexPatternSearchLayer();
+  }, [indexPattern, config.fieldName, combinedQuery]);
+
+  if (stats?.examples === undefined) return null;
```

Comment:
I don't think this is related to your PR, but for this `filebeat-apache-2019.01.30` data set, no examples are displayed for the `traefik.access.geoip.location` field. 

![image](https://user-images.githubusercontent.com/7405507/105362347-b01f8100-5bf2-11eb-88c8-04bf1a315728.png)

I remember seeing something like this in the card based version, which was related to the query we were using to get the examples - https://github.com/elastic/kibana/blob/7.9/x-pack/plugins/ml/public/application/datavisualizer/index_based/components/field_data_card/content_types/geo_point_content.tsx

If the query to get the examples returns no docs, we should probably show some sort of message in that cell saying something like `No examples were obtained for this field`

Reply:
I added this message here https://github.com/elastic/kibana/pull/88880/commits/f537b8c759fea93e1a525b3f49ddb95d44fb273d
<img width="1459" alt="Screen Shot 2021-01-25 at 10 10 23" src="https://user-images.githubusercontent.com/43350163/105732748-671f4380-5ef6-11eb-8a5e-de107c3b3552.png">

Reply:
Showing the `No examples were obtained for this field` is good here, but I think we can actually improve the search used to obtain the examples in `ml/server/models/data_visualizer/data_visualizer.ts` `getFieldExamples` too. I think we can edit that search to use the [fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html) parameter in place of the current `_source` parameter. I gave this a quick try and examples now show up for the `traefik.access.geoip.location` field above:

![image](https://user-images.githubusercontent.com/7405507/105848017-51814b00-5fd6-11eb-9853-9d663735cc05.png)

Reply:
Updated here https://github.com/elastic/kibana/pull/88880/commits/d44517fa0f5dbc9bfe854d528aa31ec5efdd1a81

The example lists are now using `fields` API, which will look different since the API auto detects and auto formats.
<img width="1781" alt="Screen Shot 2021-01-26 at 13 40 28" src="https://user-images.githubusercontent.com/43350163/105896266-3285b780-5fdc-11eb-9ce0-a9f3ae0a8998.png">

Previous format:
![](https://user-images.githubusercontent.com/7405507/105362347-b01f8100-5bf2-11eb-88c8-04bf1a315728.png)